From 8538d442b7c88158392aa16bc10fd0ad75f42a1f Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 11 Jan 2017 10:07:28 +0800
Subject: [PATCH 1467/5242] MLK-13739 video: mxsfb: use a pointer after free.

commit  0f295b50ee35f58a2378cb821fb49bdfa63ee9c0 from
https://source.codeaurora.org/external/imx/linux-imx.git

In mxsfb_overlay_exit, a pointer to freed memory is dereferenced, used as a
function argument, exchange the reference and freed function position.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 8530330..fed7c7d 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1985,8 +1985,8 @@ static void mxsfb_overlay_exit(struct mxsfb_info *fbi)
 			dma_free_writecombine(ofb->dev, ofb->video_mem_size,
 					ofb->video_mem, ofb->video_mem_phys);
 
-		framebuffer_release(ofb->ol_fb);
 		unregister_framebuffer(ofb->ol_fb);
+		framebuffer_release(ofb->ol_fb);
 	}
 }
 #else
-- 
1.7.9.5

